build.gradle.kts

```diff
@@ -3,4 +3,20 @@ plugins {
     alias(libs.plugins.android.application) apply false
     alias(libs.plugins.kotlin.android) apply false
     alias(libs.plugins.kotlin.compose) apply false
+    alias(libs.plugins.detekt)
 }
+
+
+subprojects {
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+
+    detekt {
+        config.setFrom(file("$rootDir/config/detekt/detekt.yml"))
+
+        dependencies {
+            //noinspection UseTomlInstead
+            detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.23.1")
```

Comment:
I had the same problem here, using libs version catalog throws an execption at building time. In my case it has been solved using the toml version catalog via a variable like this:

```
val detektFormatting = libs.detekt.formatting

subprojects {
    apply(plugin = "io.gitlab.arturbosch.detekt")
    detekt {
        config.setFrom(files("$rootDir/config/detekt/detekt.yml"))

        dependencies {
            detektPlugins(detektFormatting)
        }
    }
}
```